cherrypy/lib/httputil.py

```diff
@@ -202,7 +203,12 @@ def qvalue(self):
         val = self.params.get('q', '1')
         if isinstance(val, HeaderElement):
             val = val.value
-        return float(val)
+        try:
+            return float(val)
+        # Fail client requests with invalid quality value
+        # https://github.com/cherrypy/cherrypy/issues/1370
+        except ValueError:
+            raise cherrypy.HTTPError(400)
```

Comment:
I think, I'd like to see 'Malformed HTTP header' as a reason argument of the exception.

Reply:
Thanks for the quick feedback, @webknjaz.  I'll start working on a more general purpose solution.